client/client-lib/src/ln/outgoing.rs

```diff
@@ -35,3 +36,18 @@ impl OutgoingContractAccount {
         }
     }
 }
+
+#[derive(Debug, Encodable, Decodable)]
+pub struct OutgoingPaidInvoice {
+    /// The invoice itself
+    pub invoice: Invoice,
+}
+
+impl Serialize for OutgoingPaidInvoice {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(self.invoice.to_string().as_str())
+    }
+}
```

Comment:
Could we not save the invoice directly? Because it doesn't implement `Serialize` or something?

Reply:
I didn't try that before as I was following the `ConfirmedInvoice` approach, but I've just tried it now and it seems to work so will make this change.

Reply:
This has now been updated so that the invoice is saved directly.